frontend/src/pages/PredictedDataPage/PredictedDataPage.tsx

```diff
@@ -44,80 +57,122 @@ const PredictedDataPage = function () {
-        fitBitFiles?.length !== 0
-            ? fitBitFiles.map((file: FileData) => ({
-                  ...file,
-                  watch: WatchType.FITBIT,
-                  name: `${WatchType.FITBIT} ${file.id.toString()}`,
-                  isSelected: false,
-              }))
-            : [];
-
-    const availableFiles: Array<PredictedFileWithType> =
-        fitbitPredictedFiles.concat(appleWatchPredictedFiles);
+        const appleWatchFiles = [
+            {
+                id: 123,
+                data: new Uint8Array([1, 2]),
+                predictionType: null,
+                dateTime: new Date(),
+            },
+            {
+                id: 456,
+                data: new Uint8Array([5, 6]),
+                predictionType: null,
+                dateTime: new Date(),
+            },
+        ];
+        // #endregion
+
+        const appleWatchPredictedFiles: Array<PredictedFileWithType> =
+            appleWatchFiles?.length !== 0
+                ? appleWatchFiles.map((file: FileData) => ({
+                      ...file,
+                      watch: WatchType.APPLE_WATCH,
+                      name: `${WatchType.APPLE_WATCH} ${file.id.toString()}`,
+                      isSelected: false,
+                  }))
+                : [];
+
+        const fitbitPredictedFiles: Array<PredictedFileWithType> =
+            fitBitFiles?.length !== 0
+                ? fitBitFiles.map((file: FileData) => ({
+                      ...file,
+                      watch: WatchType.FITBIT,
+                      name: `${WatchType.FITBIT} ${file.id.toString()}`,
+                      isSelected: false,
+                  }))
+                : [];
+
+        return appleWatchPredictedFiles.concat(fitbitPredictedFiles);
+    };
+
+    setAvailableFiles(getAvailableFiles());
 
     /**
-     * mapping avaliable files to html components
+     * Request available files list from API
+     */
+    const refreshAvailableFiles = () => {
+        setAvailableFiles(getAvailableFiles());
+    };
+
+    /**
+     * Method Called by Refresh Button
+     */
+    const refreshFileList = () => {
+        refreshAvailableFiles();
+    };
+
+    /**
+     * Pre-conditions: A file is selected
+     * Post-conditions: Downloads the currently selected files to the users computer
+     */
+    const downloadFiles = (event: React.MouseEvent) => {
+        event.preventDefault();
+
+        // make sure at least one file is selected before you attempt to download it
+        // assert(selectedFiles.some((file: PredictedFileWithType) => file.isSelected));
+
+        selectedFiles.forEach((file: PredictedFileWithType) => {
+            if (file.isSelected) {
+                const { id, watch } = file;
+                const stringID = id.toString();
+                handleDownload(stringID, DownloadType.PREDICT, watch);
+
+                if (useDownloadError) {
+                    rollbar.error(useDownloadError);
+                }
```

Comment:
Just want to mention a better way to handle errors. I created a general component called ErrorSnackbar, it automatically handles any errors and displays it to the user. Along with  it, it also logs errors to rollbar. All you need to do is pass useDownloadError into the component

Take a look at this file for example on how it is used
https://github.com/UniversityOfSaskatchewanCMPT371/term-project-2024-team-3/blob/development/frontend/src/pages/FileUploadPage/components/UploadedFiles.tsx#L121-L123

Reply:
oh this is nice I will add that right away